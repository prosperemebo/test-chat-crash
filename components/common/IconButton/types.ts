import { ImageSourcePropType } from 'react-native'

export interface IconButtonProps {
  color?: string
  iconColor?: string
  onPress?: () => void
  disabled?: boolean
  blur?: boolean
  name?: string
  type?: 'material' | 'image'
  size?: number
  padding?: number
  source?: ImageSourcePropType
  actionNeeded?: boolean
}