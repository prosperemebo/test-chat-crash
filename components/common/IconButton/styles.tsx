import styled from 'styled-components/native'
import MaterialCommunityIcon from '@expo/vector-icons/MaterialCommunityIcons'
import { BlurView } from 'expo-blur'
import { Image } from 'expo-image'

type ButtonType = {
  color?: string
}

type IconType = {
  iconColor?: string
  size?: number
  padding?: number
  color?: number
}

type ImageIconWrapperType = {
  size?: number
  padding?: number
}

export const BlurIconButton = styled.TouchableOpacity<ButtonType>`
  border-radius: 40px;
  ${props => props.color ? `background-color: ${props.color};` : ''}
  overflow: hidden;
  width: 100%;
  height: 100%;
`

export const BlurIconButtonWrapper = styled.View`
  border-radius: 60px;
  overflow: hidden;
  flex-direction: row;
  justify-content: center;
  align-content: center;
  width: 100%;
  height: 100%;
`

export const MaterialIcon = styled(MaterialCommunityIcon) <IconType>`
  color: ${props => props.iconColor || props.theme.common.iconButtonColor};
  font-size: ${props => props.size || '27'}px;
  padding: ${props => typeof props.padding === 'number' ? props.padding : 10}px;
  ${props => props.color ? `background-color: ${props.color};` : ''}
  position: absolute;
`

export const BlurBlock = styled(BlurView)`
    height: 100%;
    width: 100%;
`

export const ImageIconWrapper = styled.View<ImageIconWrapperType>`
  height: ${props => props.size || '27'}px;
  width: ${props => props.size || '27'}px;
  padding: ${props => props.padding || 10}px;
  position: absolute;
`

export const ImageIcon = styled(Image)`
  height: 100%;
  width: 100%;
`

export const ActionBadge = styled.View`
  position: absolute;
  right: 5px;
  z-index: 10;
  height: 7px;
  width: 7px;
  background-color: ${props => props.theme.common.inputTextError};
  border-radius: 5px;
`
