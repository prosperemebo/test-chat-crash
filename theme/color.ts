export const WHITE = '#FFFFFF'
export const t10WHITE = 'rgba(255,255,255,0.1)'
export const t76WHITE = 'rgba(255,255,255,0.76)'

export const RED = '#f60004'
export const RED1 = '#D33E41'


export const BLACK = '#000000'
export const BLACK2 = '#050608'
export const nBLACK = '#15161A'
export const nBLACK10 = '#1F2025'
export const nBLACK30 = '#28292E'
export const nBLACK40 = '#050608'

export const nGRAY10 = '#6B6E79'
export const nGRAY20 = '#858995'
export const nGRAY25 = '#54575e'
export const nGRAY30 = '#2f3137'
export const nGRAY35 = '#52555D'
export const nGRAY40 = '#35363B'


export const tGRAY20 = '#85899599'
export const tGRAY30 = '#545458A6'
export const tGRAY40 = '#45464C'
export const GRAY40 = '#cfcfd0'
export const GRAY45 = '#c9c9cb'
export const GRAY50 = '#aaaaaa'
export const GRAY60 = '#52545D'
export const t40GRAY60 = '#52545D66'
export const GRAY70 = '#383B40'
export const GRAY90 = '#27292d'
export const GRAY100 = '#1f2023'
export const GRAY110 = '#151619'

export const GRAY_ONBOARD = '#8791a5'

export const BLUE = '#50b5e8'
export const BLUE2 = '#0A84FF'

export const PURPLE = '#9A1F99'

export const nBLUE = '#8D88A8'
export const nPURPLE = '#C47AF5'
export const nPURPLE10 = '#873FBB'
export const nPURPLE20 = '#873FBB'
export const nPURPLE40 = '#C882F8'

export const nORANGE10 = '#FF6705'
export const nORANGE20 = '#FF7905'
export const nORANGE40 = '#FFB700'

export const YELLOW2 = '#EBC789'
export const nYELLOW = '#F5D193'

export const nGRAYBACKGROUND2 = '#292A30'
export const nGRAYBACKGROUND3 = '#383940'
export const nGRAYBACKGROUND4 = '#989898'

export const gradient = {
  COLORS: [BLUE, PURPLE],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}

export const orangeGradient = {
  COLORS: [nORANGE10, nORANGE40],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}