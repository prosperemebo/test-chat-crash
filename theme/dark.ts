import {
  WHITE,
  BLACK,
  nGRAY10,
  GRAY_ONBOARD,
  nPURPLE,
  nBLUE,
  GRAY40,
  nGRAY20,
  RED1,
  tGRAY20,
  RED,
  nBLACK,
  nORANGE10,
  nORANGE40,
  nPURPLE10,
  GRAY70,
  tGRAY30,
  BLUE2,
  BLACK2,
  nGRAY25,
  nBLACK30,
  GRAY50,
  nBLACK10,
  t10WHITE,
  nGRAYBACKGROUND2,
  GRAY100,
  GRAY60,
  nGRAY40,
  nGRAY30,
  YELLOW2,
  nYELLOW,
  nGRAYBACKGROUND3,
  tGRAY40,
  nGRAY35,
  nBLACK40,
  t40GRAY60,
  nPURPLE20,
  nPURPLE40,
} from './color'

export const dark = {
  tint: nPURPLE,
  primaryTextColor: WHITE,
  primaryBg: nBLACK10,
  headerBg: GRAY100,
  headerBorderColor: GRAY60,
  secondaryTextColor: GRAY60,
  secondaryBgLight: nBLACK30,
  secondaryBg: nBLACK,
  serviceBg: GRAY100,
  inActiveBorderColor2: t10WHITE,
  backgroundColor2: nGRAYBACKGROUND2,
  backgroundColor3: nGRAYBACKGROUND3,
  backgroundColor4: nBLACK10,

  auth: {
    socialIconColor: WHITE,
    socialLoginButton: BLACK,
    dividerColor: nGRAY10,
    noticeText: WHITE,
    forgotPasswordLabel: WHITE,
  },

  onboarding: {
    headerTextColor: WHITE,
    headerStepsTextColor: GRAY_ONBOARD,
    progressBarColor: nPURPLE,
    progressBarBackgroundColor: nBLUE,
    selectedItemTextColor: WHITE,
    disabledItemTextColor: GRAY40,
    gameSectionOverlayColor: nPURPLE,
    gameEmptySectionColor: BLACK,
    gameEmptyIconColor: BLACK2,
  },

  common: {
    labelColor: WHITE,
    infoTextColor: nGRAY20,
    inputIconColor: nGRAY20,
    inputPlaceholderColor: nGRAY20,
    serviceTextColor: WHITE,
    inputTextColor: WHITE,
    inputTextError: RED1,
    inputBorderColor: tGRAY20,
    inputErrorBorderColor: RED,
    inputBackgroundColor: nBLACK,
    toastBackground: nBLACK,
    toastBorderColor: nPURPLE,
    toastTextColor: WHITE,
    buttonGradientOrangeStart: nORANGE10,
    buttonGradientOrangeEnd: nORANGE40,
    buttonGradientPurpleStart: nPURPLE40,
    buttonGradientPurpleEnd: nPURPLE20,
    buttonGradientGrayStart: nGRAY10,
    buttonGradientGrayEnd: nGRAY20,
    buttonDartBackgroundColor: BLACK,
    backButtonColor: nGRAY20,
    switchTextColor: WHITE,
    checkboxIconColor: nPURPLE,
    gradientButtonColorStart: nPURPLE10,
    gradientButtonColorEnd: nPURPLE,
    borderColor: GRAY70,
    genderPickerBorderColor: nGRAY25,
    genderPickerActiveTextColor: WHITE,
    genderPickerTextColor: GRAY50,
    checkboxTextColor: WHITE,
    iconButtonColor: WHITE,
    backgroundColor: nBLACK10,
  },

  settings: {
    socialIconBackgroundColor: nGRAY40,
    premiumFeatureTextColor: nPURPLE,
    buttonIconColor: nGRAY20,
    serviceItemBgColor: nGRAY30,
  },

  postOnboarding: {
    backgroundColor: nBLACK10,
    inputIconColor: nGRAY20,
    inputIconErrorColor: RED1,
    disclosureTextColor: nGRAY20,
    loadingIndicatorColor: BLACK,
  },

  profile: {
    backgroundColor: nBLACK10,
    improveBackgroundColor: YELLOW2,
    improveBannerBackgroundColor: nYELLOW,
    improveBannerTitleColor: nBLACK30,
    coverPictureCardRemoveButtonBg: tGRAY30,
    profilePicturesCountColor: tGRAY40,
  },

  chats: {
    iconColor: nGRAY20,
    inputColor: nBLACK40,
    placeholderColor: nGRAY35,
    appbarBorderColor: t40GRAY60,
    backgroundColor: nBLACK10,
  },

  nav: {
    bottomTabLabelBackgroundColor: nBLACK,
    bottomTabBackgroundColor: nBLACK,
  },

  other: {
    updateApp: {
      textBorderColor: tGRAY30,
      buttonTextColor: BLUE2,
      textColor: WHITE,
    },
  },
}
