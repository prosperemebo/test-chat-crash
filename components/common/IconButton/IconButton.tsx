import { BlurIconButton, BlurIconButtonWrapper, MaterialIcon, BlurBlock, ImageIconWrapper, ImageIcon } from './styles'
import { IconButtonProps } from './types'

const IconButton = (props: IconButtonProps) => {
  const {
    color, iconColor, onPress, disabled, blur,
    type = 'material', name, size, padding, source
  } = props

  const handlePress = () => {
    onPress && onPress()
  }

  return <BlurIconButton color={color} onPress={handlePress} disabled={disabled}>
    <BlurIconButtonWrapper>
      {blur && <BlurBlock />}
      {(type === 'material' && name) && <MaterialIcon name={name as 'image'} iconColor={iconColor} size={size} padding={padding} />}
      {type === 'image' && (
        <ImageIconWrapper size={size} padding={padding}>
          <ImageIcon source={source} contentFit={'contain'} />
        </ImageIconWrapper>
      )}
    </BlurIconButtonWrapper>
  </BlurIconButton>
}

export default IconButton