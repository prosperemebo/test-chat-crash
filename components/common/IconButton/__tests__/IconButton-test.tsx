import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { describe, expect, test } from '@jest/globals'
import IconButton from '../'
import { dark } from '@/theme'

const testIcon = require('@/assets/images/notif_icon.png')

describe('<IconButton />', () => {

  test('IconButton renders correctly', () => {
    const tree = render(
      <ThemeProvider theme={dark}>
        <IconButton type="material" name='chevron-right' color={'#FFFFFF'} iconColor={'#000000'} disabled={true} padding={0} size={30}/>
        <IconButton name={'eye-outline'} blur={true} onPress={() => {}} disabled={false} type={'material'} />
        <IconButton type='image' source={testIcon} size={50} color={'#000000'} padding={17} disabled={true} />
      </ThemeProvider>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})