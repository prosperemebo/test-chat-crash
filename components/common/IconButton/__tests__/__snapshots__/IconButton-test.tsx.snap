// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> IconButton renders correctly 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 40,
        "borderBottomRightRadius": 40,
        "borderTopLeftRadius": 40,
        "borderTopRightRadius": 40,
        "height": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignContent": "center",
          "borderBottomLeftRadius": 60,
          "borderBottomRightRadius": 60,
          "borderTopLeftRadius": 60,
          "borderTopRightRadius": 60,
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        iconColor="#000000"
        padding={0}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 30,
            },
            {
              "color": "#000000",
              "fontSize": 30,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "position": "absolute",
            },
            {
              "fontFamily": undefined,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        󰅂
      </Text>
    </View>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderBottomLeftRadius": 40,
        "borderBottomRightRadius": 40,
        "borderTopLeftRadius": 40,
        "borderTopRightRadius": 40,
        "height": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignContent": "center",
          "borderBottomLeftRadius": 60,
          "borderBottomRightRadius": 60,
          "borderTopLeftRadius": 60,
          "borderTopRightRadius": 60,
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <ViewManagerAdapter_ExpoBlurView
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
      />
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 12,
            },
            {
              "color": "#FFFFFF",
              "fontSize": 27,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
              "position": "absolute",
            },
            {
              "fontFamily": undefined,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        󰛐
      </Text>
    </View>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#000000",
        "borderBottomLeftRadius": 40,
        "borderBottomRightRadius": 40,
        "borderTopLeftRadius": 40,
        "borderTopRightRadius": 40,
        "height": "100%",
        "opacity": 1,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignContent": "center",
          "borderBottomLeftRadius": 60,
          "borderBottomRightRadius": 60,
          "borderTopLeftRadius": 60,
          "borderTopRightRadius": 60,
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <View
        padding={17}
        size={50}
        style={
          {
            "height": 50,
            "paddingBottom": 17,
            "paddingLeft": 17,
            "paddingRight": 17,
            "paddingTop": 17,
            "position": "absolute",
            "width": 50,
          }
        }
      >
        <Image
          contentFit="contain"
          source={1}
          style={
            {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </View>,
]
`;
